packages/blocks/src/page-block/edgeless/controllers/tools/shape-tool.ts

```diff
@@ -148,17 +160,57 @@ export class ShapeToolController extends EdgelessToolController<ShapeTool> {
 
     const element = this._service.getElementById(id);
     assertExists(element);
+
     this._edgeless.tools.switchToDefaultMode({
       elements: [element.id],
       editing: false,
     });
   }
 
-  onPressShiftKey(pressed: boolean) {
-    const id = this._draggingElementId;
-    if (!id) return;
+  onPressShiftKey(_pressed: boolean) {}
```

Comment:
I apologize for overlooking the scenario where holding the shift key while the mouse is stationary should adjust the shape proportionally. Upon further investigation, I've identified an issue related to redundant calls to the resize function within the `onPressedShiftKey`. Currently, this function is triggered continuously as long as the shift key is held down, leading to unnecessary calls to the resize function. In this specific use case, only the initial call is essential.

To address this, I recommend modifying the code to ensure that when using the shift key while moving, the resize function is invoked only from the `onPressShiftKey` and not redundantly from `onContainerDragMove`. If this behavior is intentional and doesn't cause any issues, it may be acceptable.
```ts
if(!ev.repeat) {
  if (event.key.toLowerCase() === 'shift' && event.shiftKey) {
    edgeless.slots.pressShiftKeyUpdated.emit(true);
  } else {
    edgeless.slots.pressShiftKeyUpdated.emit(false);
  }
}

```
It might be beneficial to include a check for `event.repeat` to ensure that the function is only called during the initial press. I believe the initial call suffices. The same approach is being used for the spacebar, and it seems to work effectively without the need for extra function calls. If you deem it necessary, I can reintroduce those additional calls.


https://github.com/toeverything/blocksuite/assets/123532141/865d429a-1946-4c5f-99e1-bd782b677a04